x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/hooks/use_missing_encryption_key_callout_has_been_dismissed.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect } from 'react';
+
+const LOCAL_STORAGE_KEY = 'fleet.missingEncryptionKeyCalloutHasBeenDismissed';
+
+export const useMissingEncryptionKeyCalloutHasBeenDismissed = (): [
+  boolean,
+  (val: boolean) => void
+] => {
+  const [isEncryptionKeyCalloutDismissed, setIsEncryptionKeyCalloutDismissed] = useState(false);
+
+  useEffect(() => {
+    const storageValue = localStorage.getItem(LOCAL_STORAGE_KEY);
```

Comment:
Can we use the `storage` service like we do in responder console? https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/components/console/components/console_state/hooks/use_stored_input_history.ts#L59-L77

I see it defined at Fleet plugin: https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/public/plugin.ts#L187

Reply:
do you mean calling the storage from the start services or did you mean returning a callback

Reply:
Calling the storage from the start service, so we use an existing implementation.